When a summary observation uses the restart keyword we don't check that timemap or refcase is present

```python
def test_that_time_map_o_refcase_is_present_if_restart_is_used_for_summary_observation():
    with pytest.raises(ConfigValidationError, match="either TIME_MAP or REFCASE"):
        ErtConfig.from_dict(
            {
                "ECLBASE": "ECLIPSE_CASE",
                "OBS_CONFIG": (
                    "obsconf",
                    [
                        (
                            "SUMMARY_OBSERVATION",
                            "FOPR",
                            {
                                "KEY": "FOPR",
                                "RESTART": "1",
                                "VALUE": "1.0",
                                "ERROR": "0.1",
                            },
                        ),
                    ],
                ),
            }
        )

```
this results in
```python
    @classmethod
    def _handle_summary_observation(
        cls,
        summary_dict: SummaryValues,
        obs_key: str,
        time_map: list[datetime],
        has_refcase: bool,
    ) -> dict[str, ObsVector]:
        summary_key = summary_dict.key
        value, std_dev = cls._make_value_and_std_dev(summary_dict)
    
        try:
            if summary_dict.date is not None and not time_map:
                # We special case when the user has provided date in SUMMARY_OBS
                # and not REFCASE or time_map so that we don't change current behavior.
                try:
                    date = datetime.fromisoformat(summary_dict.date)
                except ValueError as err:
                    raise ValueError("Please use ISO date format YYYY-MM-DD.") from err
                restart = None
            else:
                restart = cls._get_restart(summary_dict, obs_key, time_map, has_refcase)
>               date = time_map[restart]
                       ^^^^^^^^^^^^^^^^^
E               IndexError: list index out of range
```

--- a/src/ert/config/_create_observations.py
+++ b/src/ert/config/_create_observations.py
@@ -290,6 +290,11 @@ def _handle_summary_observation(
     value = summary_dict.value
     std_dev = float(_handle_error_mode(np.array(value), summary_dict))
 
+    if summary_dict.restart and not (time_map or has_refcase):
+        raise ObservationConfigError.with_context(
+            "Keyword 'RESTART' requires either TIME_MAP or REFCASE", context=obs_key
+        )
+
     if summary_dict.date is not None and not time_map:
         # We special case when the user has provided date in SUMMARY_OBS
         # and not REFCASE or time_map so that we don't change current behavior.

--- a/tests/ert/unit_tests/config/test_ert_config.py
+++ b/tests/ert/unit_tests/config/test_ert_config.py
@@ -2479,3 +2479,27 @@ def test_that_the_eclbase_keyword_sets_the_eclbase_substitution(eclbase_substitu
         ).substitutions[eclbase_substitution]
         == datafile
     )
+
+
+def test_that_time_map_or_refcase_is_present_if_restart_is_used_for_summary_observation():  # noqa E501
+    with pytest.raises(ConfigValidationError, match="either TIME_MAP or REFCASE"):
+        ErtConfig.from_dict(
+            {
+                "ECLBASE": "ECLIPSE_CASE",
+                "OBS_CONFIG": (
+                    "obsconf",
+                    [
+                        (
+                            "SUMMARY_OBSERVATION",
+                            "FOPR",
+                            {
+                                "KEY": "FOPR",
+                                "RESTART": "1",
+                                "VALUE": "1.0",
+                                "ERROR": "0.1",
+                            },
+                        ),
+                    ],
+                ),
+            }
+        )

